packages/web3-eth/types.ts

```diff
@@ -0,0 +1,160 @@
+import { HttpRpcResponse } from 'web3-providers-http/types';
+
+export type BlockIdentifier = number | 'latest' | 'earliest' | 'pending';
+
+export type EthLog = {
+    removed: boolean;
+    logIndex: string | null;
+    transactionIndex: string | null;
+    transactionHash: string | null;
+    blockHash: string | null;
+    blockNumber: string | null;
+    address: string;
+    data: string;
+    topics: string[];
+};
+
+export type EthTransaction = {
+    from: string;
+    to?: string;
+    gas?: BigInt;
+    gasPrice?: BigInt;
+    value?: BigInt;
+    data?: string;
+    nonce?: number;
```

Comment:
For the fields marked optional, they are actually defaulted by the node the request is sent to. Here is an example excerpt from the [ETH spec](https://eth.wiki/json-rpc/API):

> gas: QUANTITY - (optional, default: 90000) Integer of the gas provided for the transaction execution. It will return unused gas.

In the doc blocks I have this default behavior mentioned [e.g.](https://github.com/ChainSafe/web3.js/blob/wyatt/rewrite/eth/packages/web3-eth/src/index.ts#L588)

However, I can add the same doc blocks to `types.ts` in case users are only looking at the types

Reply:
Addressed